js/index.js

```diff
@@ -40,3 +40,97 @@ const siteContent = {
+nav.appendChild(megaElement);
+
+let coolElement = document.createElement('a');
+coolElement.textContent = 'Cool';
+nav.prepend(coolElement);
+
+navItems = nav.querySelectorAll('a');
+navItems.forEach(item => item.style.color = 'green');
+
+// cta
+let cta = document.querySelector('.cta');
+let ctaText = document.querySelector('.cta-text');
+let ctaH1 = ctaText.querySelector('h1');
+ctaH1.textContent = siteContent.cta.h1;
+let brTag = '<br />';
+ctaH1.innerHTML = ctaH1.innerHTML.split(' ').join(brTag);
+
+let ctaButton = ctaText.querySelector('button');
+ctaButton.textContent = siteContent.cta.button;
+
+let ctaImg = document.getElementById('cta-img');
+ctaImg.setAttribute('src', siteContent.cta["img-src"]);
+
+// main content
+let mainContent = document.querySelector('.main-content');
+let featuresContent = mainContent.querySelector('.top-content').children[0];
+let featuresH4 = featuresContent.querySelector('.text-content h4');
+featuresH4.textContent = siteContent["main-content"]["features-h4"];
+
+let featuresText = featuresContent.querySelector('.text-content p');
+featuresText.textContent = siteContent["main-content"]["features-content"];
+
+let aboutContent = mainContent.querySelector('.top-content').children[1];
+let aboutH4 = aboutContent.querySelector('.text-content h4');
+aboutH4.textContent = siteContent["main-content"]["about-h4"];
+
+let aboutText = aboutContent.querySelector('.text-content p');
+aboutText.textContent = siteContent["main-content"]["about-content"];
+
+let middleImg = document.getElementById('middle-img');
+middleImg.setAttribute('src', siteContent["main-content"]["middle-img-src"]);
+
+let bottomContent = mainContent.querySelector('.bottom-content');
+let servicesContent = bottomContent.children[0];
+let servicesH4 = servicesContent.querySelector('.text-content h4');
+servicesH4.textContent = siteContent["main-content"]["services-h4"];
+
+let servicesText = servicesContent.querySelector('.text-content p');
+servicesText.textContent = siteContent["main-content"]["services-content"];
+
+let productContent = bottomContent.children[1];
+let productH4 = productContent.querySelector('.text-content h4');
+productH4.textContent = siteContent["main-content"]["product-h4"];
+
+let productText = productContent.querySelector('.text-content p');
+productText.textContent = siteContent["main-content"]["product-content"];
+
+let visionContent = bottomContent.children[2];
+let visionH4 = visionContent.querySelector('.text-content h4');
+visionH4.textContent = siteContent["main-content"]["vision-h4"];
+
+let visionText = visionContent.querySelector('.text-content p');
+visionText.textContent = siteContent["main-content"]["vision-content"];
+
+// contact
+let contact = document.querySelector('.contact');
+let contactH4 = contact.querySelector('h4');
+contactH4.textContent = siteContent.contact["contact-h4"];
+
+let address = contact.getElementsByTagName('p')[0];
+address.textContent = siteContent.contact.address;
```

Comment:
I feel you code could be DRYer if, instead of declaring a new variable for every element, you store them in an array and call them directly from there. 
It's fine like this, and for sure it's more declarative, but longer as well and in my opinion not really easier to read.
You did that on line 46 and I personally find it better.
This is just my view on it anyway.